main.go

```diff
@@ -0,0 +1,475 @@
+	if err != nil {
+		glog.Errorf("Error parsing serviceAccounts %v", err)
+		return
+	}
+
+	// Select an random serviceAccount to use.
+	impersonateAccount, svcKeyCred := getRandomServiceAccount(serviceAccounts, svcAccountKeys)
+
+	// Initialize the Cloud Asset API.
+	// This api is used to find the projects in an ORG
+	assetClient, err := getAssetClient(ctx, impersonateAccount, svcKeyCred)
+	if err != nil {
+		glog.Errorf("Error getting Cloud Asset API Client %v", err)
+		return
+	}
+
+	// Use the Cloud Asset API to recall/find the Organization for provided organizationID.
+	// We do not really need to do this section since the the orgID was provided in the input argument.
+	// We are also setting a default "emptyQuery" value here to return all projects.
+	// We are doing this as a type of input validation with the asset-api to search/narrow the organization.
+	emptyQuery := ""
+	allOrganizations, err := findResourcesByAssetType(ctx, *organization, assetTypeOrganization, emptyQuery, assetClient)
+	if err != nil {
+		glog.Errorf("Error finding Organizations %v", err)
+		return
+	}
+
+	// Use the Cloud Asset API to recall/find the projects for provided organizationID.
+	// We do not use the Resource Manager api here since that would return all the projects
+	// the caller has access to, not just those restricted to the organization we want.
+	// We are also setting a default "emptyQuery" value here to return all projects.
+	// You can specify a query filter here to configure the set of projects to evaluate.
+	// see https://cloud.google.com/asset-inventory/docs/searching-resources#how_to_construct_a_query
+	allProjects, err := findResourcesByAssetType(ctx, *organization, assetTypeProject, emptyQuery, assetClient)
+	if err != nil {
+		glog.Errorf("Error finding all projects in the organization %v", err)
+		return
+	}
+
+	glog.V(20).Infoln("      Getting Users")
+
+	// Initialize the Workspace Admin client
+	// This client will be used to find users in a given Cloud Identity/Workspace domain
+	adminService, err := getAdminServiceClient(ctx, impersonateAccount, svcKeyCred, *subject)
+	if err != nil {
+		glog.Errorf("Error initializing admin client %v", err)
+		return
+	}
+
+	// If you want to narrow the search to a subset of users, apply a searchFilter here
+	//  https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+	//	eg. searchFilter := "isAdmin=false"
+
+	searchFilter := ""
+	allUsers, err := findDomainUsers(ctx, *cx, searchFilter, adminService)
+	if err != nil {
+		glog.Errorf("Error finding domain users %v", err)
+		return
+	}
+
+	// Launch a goroutines that will process the projects and organizations
+	// that are visible to users.  The channel is fed by the
+	// getOrganizations(), and getProjects() function calls in the succeeding stanza.
+	ch := make(chan *userAccess)
+	done := make(chan bool)
+	go func() {
+		for {
+			msg, ok := <-ch
+			if ok {
```

Comment:
```suggestion
		for msg := range ch {
```

When you `range` over a channel, it stops when the channel gets closed. See https://tour.golang.org/concurrency/4.